### ISSUE TITLE
[Wave6] Migrate the Collect policy data pusher events to Onyx

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#34451]
### Problem

We use the [getPolicyOnyxUpdates](https://github.com/Expensify/Web-Expensify/blob/52be76b273c18f306223510c2ac36c7df70dd221/lib/PolicyAPI.php#L6127-L6213) method in PHP to format the necessary policy data for the Collect policy. They are needed to show the categories, default billable, next steps, scheduled submit settings, taxes, report field etc. We now can queue these updates from Auth so lets do that

###

Lets move this to Auth so we can return it directly from OpenReport or OpenApp commands, some of this data is easy to get as they are just boolean flags but some has more complicated formatting.

**Comment by mountiny** [comment]
cc @yuwenmemon @amyevans as you might have context on these

**Comment by amyevans** [comment]
Some info on the tags-related keys:
* For setting the `policyTags_` key, this is going to be similar to https://github.com/Expensify/Auth/blob/9a8621cae14ea3d1deb0ccc321594f4cb7cba621/auth/lib/Policy.cpp#L2746-L2753 (although not identical, since that is merging in the value for only one tag list instead of setting the value for all tag lists).

* For `recentlyUsedTags_` I don't believe it requires reformatting, just get the NVP value:
```cpp
JSON::Value recentlyUsedTags = NVP::getAsJSON(db, memberID, NVP::RECENTLY_USED_TAGS_PREFIX + policyID);
```

In general though I think all the constants are named the same in Auth as they are in Web-E so just doing a search in Auth for the key, e.g. `OnyxUpdates::COLLECTION_POLICY_RECENTLY_USED_TAGS` should yield any relevant code to lean on!

**Comment by thienlnam** [comment]
Report field updates are new, so I don't think there are any methods that already exist to help format them to the shape for Onyx

**Comment by yuwenmemon** [comment]
Asked a performance question about this here: https://expensify.slack.com/archives/C03TQ48KC/p1705453245561089

**Comment by yuwenmemon** [comment]
PR is in review. Pretty close to being done we're going to do some benchmarking on some large policies.

**Comment by yuwenmemon** [comment]
@mountiny Small question about this... What's the larger context for needing this change? I was talking with @iwiznia about this earlier and from what we can see both `OpenReport` and `AppInit` are READ operations that wouldn't require sending a push notification update to all clients with the latest policy data. Is this just a matter of moving logic from PHP to Auth or does this grant us some sort of extra functionality in the future?

**Comment by mountiny** [comment]
@yuwenmemon Correct, that was a bad wording on used in the issue title/description. Its not passed using pusher, but in the onyxUpdate format sent as a response of Read call. I will update this to make it clear.

I believe we have discussed this before, but correct this is mainly to move the logic to Auth so we avoid more ready Auth calls during the OpenApp API command.

**Comment by mountiny** [comment]
Updated.

**Comment by yuwenmemon** [comment]
Web PR merged!
